src/append/rolling_file/mod.rs

```diff
@@ -182,7 +180,16 @@ impl Append for RollingFileAppender {
 
         // TODO(eas): Idea: make this optionally return a future, and if so, we initialize a queue for
         // data that comes in while we are processing the file rotation.
-        self.policy.process(&mut file)
+
+        //first, rotate
+        self.policy.process(&mut file)?;
```

Comment:
This can cause a substantial slowdown if the background rolling is not enabled. We should call this out in the time rolling docs.